### PR TITLE
fix(backend): log stack trace included

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Include log stack trace

### Issues

- fixes https://github.com/gradido/gradido/issues/2357
- fixes https://github.com/gradido/gradido/pull/2751

### Todo

- [X] None

### CODE REVIEW

**Review by clauspeterhuebner — DISMISSED** [review]
Nice that you found a config feature to print stack-info on message-level.
But on my opinion is this not a productive solution as the following 
**log-output shows:**

```
2023-04-04T15:21:25.000 INFO apollo [unknown] [D:\Dev\Gradido\backend\src\server\createServer.ts : 77] -  running with PRODUCTION=false, sending EMAIL enabled=true and EMAIL_TEST_MODUS=false ...     at D:\Dev\Gradido\backend\src\server\createServer.ts:77:10
    at Generator.next (<anonymous>)
    at fulfilled (D:\Dev\Gradido\backend\src\server\createServer.ts:24:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-04-04T15:21:25.020 INFO backend [unknown] [D:\Dev\Gradido\backend\src\federation\validateCommunities.ts : 14] -  Federation: startValidateCommunities loop with an interval of 60000 ms...     at startValidateCommunities (D:\Dev\Gradido\backend\src\federation\validateCommunities.ts:14:10)
    at D:\Dev\Gradido\backend\src\index.ts:17:32
    at Generator.next (<anonymous>)
    at fulfilled (D:\Dev\Gradido\backend\src\index.ts:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Server is running at http://localhost:4000

```
**against the previous logoutput before:**

```
2023-04-04T15:21:25.000 INFO apollo [unknown] [D:\Dev\Gradido\backend\src\server\createServer.ts : 77] -  running with PRODUCTION=false, sending EMAIL enabled=true and EMAIL_TEST_MODUS=false ...
2023-04-04T15:21:25.020 INFO backend [unknown] [D:\Dev\Gradido\backend\src\federation\validateCommunities.ts : 14] -  Federation: startValidateCommunities loop with an interval of 60000 ms...     
Server is running at http://localhost:4000

```

this solution will growup our logfiles to a not acceptable size and should at least limited to only the error logger

**Comment by Elweyn** [comment]
@clauspeterhuebner this is because we call 
```
console.log('ogger.info(`Server is running at http://localhost:${CONFIG.PORT}`)
```

File: backend/log4js-config.json [hunk 1]
```diff
@@ -77,15 +77,15 @@
       "type": "stdout",
       "layout": 
       {
-        "type": "pattern", "pattern": "%d{ISO8601} %p %c [%X{user}] [%f : %l] -  %m"
+        "type": "pattern", "pattern": "%d{ISO8601} %p %c [%X{user}] [%f : %l] -  %m %s"
```
Comment (clauspeterhuebner) [inline backend/log4js-config.json:80]: i would preferre a defensive solution to enable the stack-output only for error cases.
the layout-pattern for appenders out and appoloOut should not use the %s for stack-output, even when it seems to generate logoutput only on console output. 
this has to be ensured before activating it on master-branch and the log-files will increase in an unexpected manner.

File: backend/log4js-config.json [hunk 2]
```diff
@@ -77,15 +77,15 @@
...
       }
     },
     "apolloOut": 
     {
       "type": "stdout",
       "layout": 
       {
-        "type": "pattern", "pattern": "%d{ISO8601} %p %c [%X{user}] [%f : %l] -  %m"
+        "type": "pattern", "pattern": "%d{ISO8601} %p %c [%X{user}] [%f : %l] -  %m %s"
```
Comment (clauspeterhuebner) [inline backend/log4js-config.json:88]: i would preferre a defensive solution to enable the stack-output only for error cases.
the layout-pattern for appenders out and appoloOut should not use the %s for stack-output, even when it seems to generate logoutput only on console output. 
this has to be ensured before activating it on master-branch and the log-files will increase in an unexpected manner.

**Review by clauspeterhuebner — DISMISSED** [review]

**Pull request merged**